Sources/LanguageServerProtocol/SupportTypes/DocumentURI.swift

```diff
@@ -69,18 +56,29 @@ public struct DocumentURI: Codable, Hashable {
     guard let url = URL(string: string) else {
       fatalError("Failed to construct DocumentURI from '\(string)'")
     }
-    self.init(standardize(url: url))
+    self.init(url)
   }
 
   public init(_ url: URL) {
     self.storage = url
+    self.standardizedStorage = url.standardizedFileURL
     assert(self.storage.scheme != nil, "Received invalid URI without a scheme '\(self.storage.absoluteString)'")
   }
 
   public init(from decoder: Decoder) throws {
     self.init(string: try decoder.singleValueContainer().decode(String.self))
   }
 
+  /// Equality check to handle escape sequences in URLs, which we otherwise preserve.
```

Comment:
I tried using `self.storage.standardizedFileURL` instead and it didn't work, I think something there must be mutable. Also considering instead using `stringByRemovingPercentEncoding` from NSString